#!/usr/bin/env python
PACKAGE = "egm_path"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("conv_threshold_trans",     double_t,      0,  "Convergence threshold for the translation [mm]", 3., 0., 5000)
gen.add("conv_threshold_rot",       double_t,      0,  "Convergence threshold for the rotation [deg]", 0.5, 0., 360.)
gen.add("gain_trans_p",             double_t,      0,  "Translation proportional gain [-]",             0.8, 0., 2.)
gen.add("gain_trans_i",             double_t,      0,  "Translation integral gain [-]",                 0.1, 0., 1.)
gen.add("gain_rot_p",               double_t,      0,  "Rotation proportional gain [-]",                0.2, 0., 1.)
gen.add("beta_LPF",                 double_t,      0,  "Low Pass Filter beta value [-]",                0.8, 0.1, 1.)
gen.add("verbose",                  bool_t,        0,  "Print additional information",  True)

exit(gen.generate(PACKAGE, "egm_path", "EGMPathParams"))
